src/utils.ts

```diff
@@ -409,7 +409,7 @@ export async function logDuration<T>(logger: BaseLogger, name: string, block: ()
  * Promise/async version of {@link setImmediate}.
  */
 export function immediate(): Promise<void> {
-    return new Promise(setImmediate);
+    return new Promise((resolve) => setTimeout(resolve));
```

Comment:
(aside: given this doesn't work any more in modern browsers, how did we get this far without fixing it? Is there a polyfill somewhere we can disable?)

Reply:
Node supports it, EW loads a polyfill for both runtime & jest tests. See related PRs
![image](https://github.com/matrix-org/matrix-js-sdk/assets/2403652/3129ac6e-42e5-41fd-b4ed-6416c93f395d)